src/Tools/dotnet-watch/src/dotnet-watch.csproj

```diff
@@ -22,13 +22,42 @@
   <ItemGroup>
     <Reference Include="Microsoft.AspNetCore" />
     <Reference Include="Microsoft.AspNetCore.WebSockets" />
+
+    <ProjectReference
+      Include="..\BrowserRefresh\src\Microsoft.AspNetCore.Watch.BrowserRefresh.csproj"
+      PrivateAssets="All"
+      ReferenceOutputAssembly="false"
+      SkipGetTargetFrameworkProperties="true"
+      UndefineProperties="TargetFramework;TargetFrameworks" />
   </ItemGroup>
 
+  <Target Name="_CopyMiddlewareForLocal" BeforeTargets="PrepareForRun">
+    <PropertyGroup>
+      <BrowserMiddlewarePath>$(ArtifactsBinDir)Microsoft.AspNetCore.Watch.BrowserRefresh\$(Configuration)\netcoreapp3.1\Microsoft.AspNetCore.Watch.BrowserRefresh.dll</BrowserMiddlewarePath>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <None Include="$(BrowserMiddlewarePath)" Pack="true" PackagePath="tools\$(DefaultNetCoreTargetFramework)\any\middleware"/>
+    </ItemGroup>
+
+    <Copy
+      SourceFiles="$(BrowserMiddlewarePath)"
+      DestinationFolder="$(OutputPath)middleware"
+      OverwriteReadOnlyFiles="true"
+      SkipUnchangedFiles="true"/>
+  </Target>
+
+  <Target Name="PopulateNuspec" BeforeTargets="InitializeStandardNuspecProperties">
+    <ItemGroup>
+      <None Include="$(ArtifactsBinDir)Microsoft.AspNetCore.Watch.BrowserRefresh\$(Configuration)\netcoreapp3.1\Microsoft.AspNetCore.Watch.BrowserRefresh.dll" Pack="true" PackagePath="middleware"/>
+    </ItemGroup>
+  </Target>
+
   <Target Name="_FixupRuntimeConfig" BeforeTargets="_GenerateRuntimeConfigurationFilesInputCache">
     <ItemGroup>
       <_RuntimeFramework Include="@(RuntimeFramework)" />
       <RuntimeFramework Remove="@(RuntimeFramework)" />
-      <RuntimeFramework Include="Microsoft.AspNetCore.App" FrameworkName="Microsoft.AspNetCore.App" Version="5.0.0-preview" />
+      <RuntimeFramework Include="Microsoft.AspNetCore.App" FrameworkName="Microsoft.AspNetCore.App" Version="5.0.0-preview.7.20329.4" />
```

Comment:
Why this specific version instead of letting it float? This looks odd outside eng/Versions.props in any case.

Reply:
I changed this back to the earlier version. I had an earlier build of the runtime locally which didn't have some of the newer runtime types such as the non-generic `TaskCompletionSource`. I need to sort this out soon for rc1 since it also affects blazor